cmd/kubelet/app/server.go

```diff
@@ -598,6 +623,72 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) {
+	pool := x509.NewCertPool()
+	if len(clientConfig.CAData) > 0 {
+		if certs, err := certutil.ParseCertsPEM(clientConfig.CAData); err != nil {
+			glog.Warningf("Unable to load the CA from data.")
+		} else {
+			for _, cert := range certs {
+				pool.AddCert(cert)
+			}
+		}
+	}
+	if len(clientConfig.CAFile) > 0 {
+		if certs, err := certutil.NewPool(clientConfig.CAFile); err != nil {
+			glog.Warningf("Unable to load the CA from file.")
+		} else {
+			pool = certs
```

Comment:
Since this is overwriting the CAData certs, should you add validation that only 1 is set?